plugins/org.jkiss.dbeaver.model.ai/src/org/jkiss/dbeaver/model/ai/engine/copilot/CopilotClient.java

```diff
@@ -221,6 +225,64 @@ public Flow.Publisher<CopilotChatChunk> createChatCompletionStream(
         return publisher;
     }
 
+    /**
+     * Retrieves a list of Copilot model IDs. If the current model list is empty or if a force refresh
+     * is required, it attempts to fetch and load models using the provided authentication token.
+     *
+     * @param monitor the progress monitor to report task progression and check for task cancellation
+     * @param token an authentication token used to fetch the models; can be null
+     * @param forceRefresh a boolean flag indicating whether to force a refresh of the model list
+     * @return a list of model IDs as strings; an empty list is returned if no models are available
+     */
+    public static List<String> getModels(@NotNull DBRProgressMonitor monitor, @Nullable String token, boolean forceRefresh) {
+        if ((models.isEmpty() || forceRefresh) && CommonUtils.isNotEmpty(token)) {
+            try (CopilotClient copilotClient = new CopilotClient()) {
+                copilotClient.loadModels(monitor, token);
+            } catch (Exception ex) {
+                DBWorkbench.getPlatformUI().showError(
+                    "Error reading model list",
+                    "Failed to read the model list",
+                    ex
+                );
+            }
+        }
+        if (models.isEmpty()) {
+            return List.of();
+        }
+        return models.values().stream().map(CopilotModel::id).toList();
+    }
+
+    /**
+     * Loads the available Copilot models by making an HTTP request to a specified endpoint.
+     * Filters the models by chat-specific criteria before returning them.
+     *
+     * @param monitor the progress monitor to check for task cancellation and update task progress
+     * @param token the authentication token required to authorize the HTTP request
+     * @return a list of chat-enabled Copilot models
+     * @throws DBException if the request fails or encounters an error during execution
+     */
+    public List<CopilotModel> loadModels(@NotNull DBRProgressMonitor monitor, @NotNull String token) throws DBException {
+        HttpRequest request = HttpRequest.newBuilder()
+            .header("Content-type", "application/json")
+            .uri(AIHttpUtils.resolve(COPILOT_CHAT_MODELS_URL))
+            .header("authorization", "Bearer " + token)
+            .header("Editor-Version", CHAT_EDITOR_VERSION)
+            .GET()
```

Comment:
The ordering is all mixed up

```suggestion
            .GET()
            .uri(AIHttpUtils.resolve(COPILOT_CHAT_MODELS_URL))
            .header("Content-type", "application/json")
            .header("authorization", "Bearer " + token)
            .header("Editor-Version", CHAT_EDITOR_VERSION)
```